lib/src/integrationTest/kotlin/com/lemonappdev/konsist/core/declaration/kofile/KoFileDeclarationForKoFunctionProviderTest.kt

```diff
@@ -99,6 +103,158 @@ class KoFileDeclarationForKoFunctionProviderTest {
+            ) shouldBeEqualTo false
+            containsFunction(
+                "sampleFunction",
+                INTERNAL,
+                PRIVATE,
+                includeNested = false,
+                includeLocal = false,
+            ) shouldBeEqualTo false
+            containsFunction(
+                "sampleLocalFunction",
+                SUSPEND,
+                includeNested = false,
+                includeLocal = true,
+            ) shouldBeEqualTo true
+            containsFunction(
+                "sampleLocalFunction",
+                SUSPEND,
+                includeNested = false,
+                includeLocal = false,
+            ) shouldBeEqualTo false
+            containsFunction(
+                "sampleLocalFunction",
+                PRIVATE,
+                includeNested = false,
+                includeLocal = true,
+            ) shouldBeEqualTo false
+            containsFunction(
+                "sampleNestedFunction",
+                OPEN,
+                includeNested = true,
+                includeLocal = false,
+            ) shouldBeEqualTo true
+            containsFunction(
+                "sampleNestedFunction",
+                OPEN,
+                includeNested = false,
+                includeLocal = false,
+            ) shouldBeEqualTo false
+            containsFunction(
+                "sampleNestedFunction",
+                PRIVATE,
+                includeNested = true,
+                includeLocal = false,
+            ) shouldBeEqualTo false
+        }
+    }
+
+    @Test
+    fun `contains-functions-with-regex`() {
+        // given
+        val sut = getSnippetFile("contains-functions-with-regex")
+            .files
+            .first()
+
+        // then
+        assertSoftly(sut) {
+            containsFunction(Regex("[a-zA-Z]+"), includeNested = false, includeLocal = false) shouldBeEqualTo true
+            containsFunction(Regex("[a-zA-Z]+"), includeNested = false, includeLocal = true) shouldBeEqualTo true
+            containsFunction(Regex("[0-9]+"), includeNested = false, includeLocal = false) shouldBeEqualTo false
+            containsFunction(Regex("[a-zA-Z]+"), includeNested = true, includeLocal = false) shouldBeEqualTo true
+            containsFunction(Regex("[0-9]+"), includeNested = false, includeLocal = false) shouldBeEqualTo false
+        }
+    }
+
+    @Test
+    fun `contains-functions-with-modifiers-and-regex`() {
+        // given
+        val sut = getSnippetFile("contains-functions-with-modifiers-and-regex")
+            .files
+            .first()
+
+        // then
+        assertSoftly(sut) {
+            containsFunction(
+                Regex("[a-zA-Z]+"),
+                INTERNAL,
+                includeNested = false,
+                includeLocal = false,
+            ) shouldBeEqualTo true
+            containsFunction(Regex("[0-9]+"), INTERNAL, includeNested = false, includeLocal = false) shouldBeEqualTo false
```

Comment:
Please update the formatting on this one nad few ones below to keep it consistent